Makefile

```diff
@@ -45,13 +58,29 @@ vet:
 # build targets
 #
 
-bin/%:
-	$(Q)echo "Building $@..."; \
-	$(GO_BUILD) -o $@ ./$(subst bin/,cmd/,$@)
+$(BINARIES): bin/%:
+	$(Q)echo "Building $@..."
+	$(Q)(cd cmd/$(*) && $(GO_BUILD) -o $(abspath $@) .)
+
+#
+# vendoring targets
+#
+.PHONY: mod-verify mod-tidy $(MOD_TIDY_TARGETS) $(MOD_VERIFY_TARGETS)
+
+MOD_TARGETS := $(CMDS) ..
+MOD_TIDY_TARGETS := $(patsubst %,mod-tidy-%,$(MOD_TARGETS))
+MOD_VERIFY_TARGETS := $(patsubst %,mod-verify-%,$(MOD_TARGETS))
+
+$(MOD_TIDY_TARGETS): mod-tidy-%:
+	$(Q)echo "Running $@... in $(abspath ./cmd/$(*))"
+	$(Q)(cd $(abspath ./cmd/$(*)) && ${GO} mod tidy)
+
+$(MOD_VERIFY_TARGETS): mod-verify-%: mod-tidy-%
```

Comment:
What is the purpose of `mod-verify-% mod-tidy-%` here ? I think they will evaluate to empty. Shouldn't this be instead
```
$(MOD_VERIFY_TARGETS): $(MOD_TIDY_TARGETS)
```

Reply:
That said, I think strictly in the `make` sense something like `mod-verify-%: mod-tidy-%` is more correct: "whenever verifying a module, tidy it up first"...

Reply:
This means that each `mod-verify-%` target depends on its specific `mod-tidy-%` target. This means that for each one `tidy` is run before `verify`.

I don't think that we need to enforce the global dependencies here i.e: before ANY verify target is run, run ALL tidy targets.

Reply:
But a simple `mod-verify-%: mod-tidy-%` would already cover that dependency. There is no need for the `$(MOD_VERIFY_TARGETS):` there in the beginning. 

And I actually misread that whole line before posting my original comment. I thought it said  `$(MOD_VERIFY_TARGETS): mod-verify-% mod-tidy-%`...

But even now with that more careful look, I have to say that I still don't understand what does that extra `$(MOD_VERIFY_TARGETS):` accomplishes there... Actually  and I wasn't even aware you could write a rule like that... or if I ever did I've long forgotten since.

Reply:
Without the initial part the pattern rules are not matched. Afk at the moment, so let me find a docs link tomorrow.

Reply:
See [static pattern rules](https://web.mit.edu/gnu/doc/html/make_4.html#SEC37)